### ISSUE TITLE
Prevent execution within %Appdata% with SRP/Applocker

### ISSUE THREAD

**Issue by mschilt** [securitywithoutborders/hardentools#54]
How about enabling Software Restriction Policy or Applocker to prevent execution from the %Appdata% path? This prevents most of the 'finger faster than brain' infections and also some of multi-stage malware.

Users will get an error message telling that something was blocked.

https://docs.microsoft.com/en-us/windows/security/threat-protection/applocker/using-software-restriction-policies-and-applocker-policies

**Comment by Dyras** [comment]
Sounds great in theory but in reality it will break so many programs it's not even worth considering.

**Comment by botherder** [comment]
Yeah, that's what I'm also afraid of. There's a lot of legitimate software that load things from AppData unfortunately.

**Comment by obsti8383** [comment]
I have the same experience. Quite effective against simple malware, but it affects quite a lot of legitimate software (also Auto-Updaters sometimes - which is really bad).

**Comment by sir-ragna** [comment]
I believe this is a good feature. This has proven to be effective at stopping malware in a lot of organisations. It is also advised by authorities in the cyber security field such as the SANS Institute.

This might block legitimate software, but that is a reality of most hardening techniques. 

If breaking other software is too great of a concern then there are a number of things that could be considered.

- Make it non-default
- Inform the user
- Show a list of programs that is known to break
- Scan the appdata for software that will break
- Give the user the option to white list certain programs (this is possible with Applocker)

It might be a challenge to make this feature user friendly, but I believe it is definitely worth making it accessible to the average person.

**Comment by botherder** [comment]
I think making it a non-default could be actually a good option.

**Comment by Dyras** [comment]
@botherder It would need a big fat warning that it will likely break auto-updating for a plethora of programs, including Wire, Brave, Chrome, Opera and Origin (?).

**Comment by botherder** [comment]
Fair point. If you guys have actually verified that it is such a popular location for auto-update, I think we should shelf this one.

**Comment by Dyras** [comment]
It's easy to check really. Go into the AppData folder and search for "exe" and you will see that a shitton of programs will appear. Hell, some programs even install to the AppData folder to make it easier to update them without UAC prompts.

**Comment by botherder** [comment]
Lovely.

**Comment by beerisgood** [comment]
@sir-ragna AppLocker isnt even in Pro edition

**Comment by sir-ragna** [comment]
@beerisgood you are correct. I messed up.

**Comment by obsti8383** [comment]
Looks like Microsoft killed Software Restriction Policies with Windows 11 version 22H2...
